front/pages/api/w/[wId]/subscriptions/index.ts

```diff
@@ -1,18 +1,32 @@
 import type { PlanType, WithAPIErrorReponse } from "@dust-tt/types";
+import { assertNever } from "@dust-tt/types";
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
 import type { NextApiRequest, NextApiResponse } from "next";
 
 import { Authenticator, getSession } from "@app/lib/auth";
+import {
+  cancelSubscriptionImmediately,
+  skipSubscriptionFreeTrial,
+} from "@app/lib/plans/stripe";
 import {
   downgradeWorkspaceToFreePlan,
   getCheckoutUrlForUpgrade,
 } from "@app/lib/plans/subscription";
 import logger from "@app/logger/logger";
 import { apiError, withLogging } from "@app/logger/withlogging";
 
-export type PostSubscriptionResponseBody = {
-  plan: PlanType;
-  checkoutUrl?: string;
-};
+export type PostSubscriptionResponseBody =
```

Comment:
NIT: PostSubscriptionResponseBody has not changed, I would introduce a PatchSubscriptionResponseBody instead (or if we want to keep one rename it to PostOrPatchSubscriptionResponseBody)